src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtil.java

```diff
@@ -300,7 +300,114 @@ public static boolean isOfType(int type, int... types) {
      * @return true if type matches one of the given types.
      */
     public static boolean isOfType(DetailAST ast, int... types) {
-        return ast != null && isOfType(ast.getType(), types);
+        return isOfType(ast.getType(), types);
+    }
+
+    /**
+     * Determines if the token type belongs to the given first parameter.
+     *
+     * @param type the Token Type to check
+     * @param first the acceptable token
+     *
+     * @return true if type equals to first
+     */
+    public static boolean isOfType(int type, int first) {
+        return type == first;
```

Comment:
Why we need this method?
Remove it

Reply:
I forgot to remove it during generating the report. Changes Done

Reply:
@romani, why should we remove this method? If we remove it, Checks will keep using 'unoptimized' `isOfType(int type, int... types)` instead, which we try to avoid in this fix.
Currently this new method `isOfType(int type, int first)` is used in UnnecessaryParenthesesCheck.

Reply:
Ok, let's keep it, but no coverage by pure Unit tests for method.

Reply:
@romani except this, We have to do coverage with pure tests only.